eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/UnifiedSet.java

```diff
@@ -1330,6 +1330,115 @@ private int chainedHashCode(ChainedBucket bucket)
         while (true);
     }
 
+    public void trimToSize()
+    {
+        Object[] temp = this.table;
+        this.allocateTable(this.occupied);
+        this.occupied = 0;
+
+        for (int j = 0; j < temp.length; j++)
+        {
+            Object cur = temp[j];
+            if (cur instanceof ChainedBucket)
+            {
+                ChainedBucket bucket = (ChainedBucket) cur;
+                this.chainedTrimToSize(bucket);
+            }
+            else if (cur != null)
+            {
+                this.addWithoutResize(this.nonSentinel(cur));
+            }
+        }
+    }
+
+    private void chainedTrimToSize(ChainedBucket bucket)
+    {
+        do
+        {
+            this.addWithoutResize(this.nonSentinel(bucket.zero));
+            if (bucket.one == null)
+            {
+                return;
+            }
+            this.addWithoutResize(this.nonSentinel(bucket.one));
+            if (bucket.two == null)
+            {
+                return;
+            }
+            this.addWithoutResize(this.nonSentinel(bucket.two));
+            if (bucket.three == null)
+            {
+                return;
+            }
+            if (bucket.three instanceof ChainedBucket)
+            {
+                bucket = (ChainedBucket) bucket.three;
+                continue;
+            }
+            this.addWithoutResize(this.nonSentinel(bucket.three));
+            return;
+        }
+        while (true);
+    }
+
+    private void addWithoutResize(T key)
+    {
+        int index = this.index(key);
+        Object cur = this.table[index];
+        if (cur == null)
+        {
+            this.table[index] = UnifiedSet.toSentinelIfNull(key);
+            this.occupied++;
+            return;
+        }
+        if (cur instanceof ChainedBucket || !this.nonNullTableObjectEquals(cur, key))
```

Comment:
I don't think you need this if statement. If cur != null, we will always do a chained add